supportedBackends/google/pipelines/v2alpha1/src/main/scala/cromwell/backend/google/pipelines/v2alpha1/PipelinesUtilityConversions.scala

```diff
@@ -22,7 +26,39 @@ trait PipelinesUtilityConversions {
     .setName(disk.name)
     .setSizeGb(disk.sizeGb)
     .setType(disk.diskType |> toV2DiskType)
-  def toExecutionEvent(event: Event) : ExecutionEvent = ExecutionEvent(event.getDescription, OffsetDateTime.parse(event.getTimestamp))
+  private def shouldPublish(event: Event): Boolean = {
+    // The Docker image used for CWL output parsing causes some complications for the timing diagram. Docker image
+    // pulling is done automatically by PAPI v2 and therefore does not correspond to any Actions generated in Cromwell.
+    // Since there are no Actions there are no labeled Actions and therefore Docker pull events do not get grouped into
+    // any labeled categories. For regular prefetched images this is okay since they don't conflict with any other labeled
+    // activity and we actually want to see that detail on the timing diagram. But the CWL output parsing image pull happens
+    // in the middle of other 'Delocalization' events. Since it is not a 'Delocalization' event it appears to the timing diagram
+    // logic to be concurrent to 'Delocalization' and another row is wrongly added to the timing diagram. The logic here checks
+    // if the event description matches a CWL Docker image pull and if so suppresses publication, which is enough to
+    // make the timing diagram happy.
+    !event.getDescription.matches(s"""(Started|Stopped) pulling "$CwlOutputJsonProcessingDockerImage"""")
+  }
+
+  def toExecutionEvent(actionIndexToEventType: Map[Int, String])(event: Event): Option[ExecutionEvent] = {
+    val groupingFromAction = for {
+      rawValue <- Option(event.getDetails.get("actionId"))
+      integerValue <- Try(Integer.valueOf(rawValue.toString)).toOption
+      group <- actionIndexToEventType.get(integerValue)
+    } yield group
+
+    // There are both "Started pulling" and "Stopped pulling" events but these are confusing for metadata, especially on the
+    // timing diagram. Create a single "Pulling <docker image>" grouping to absorb these events.
+    def groupingFromPull: Option[String] = List("Started", "Stopped") flatMap { k =>
```

Comment:
OOC any particular reason why this one is a `def` and the one above a `val` ?

Reply:
Also TOL: why is the grouping for this one done here instead of during the metadata building like for the other ones ?

Reply:
The `val` vs `def` is because the first one always has to run but the second is the "else" of an `orElse`. The grouping logic here is different from the grouping that's currently in the metadata building, but I think this could still be moved over there to keep all the grouping logic in one place.

Reply:
Oh I remember now...this is all PAPI v2 specific filtering and the MetadataBuilder is meant to be generic.

Reply:
Welp those BCBIO changes will certainly necessitate some refactoring here. 😛 I'm inclined to have the "remove docker pull events during Delocalization" here in the PAPI v2 part of the world since the details of that really are specific to PAPI v2.

Reply:
Oh yeah sorry about the forced rebase -_- Yes I think keep that here is totally fine